src/tensor_ops/select_and_gather/mod.rs

```diff
@@ -157,9 +160,9 @@ pub trait GatherTo<D: DeviceStorage>: HasErr + HasShape {
 impl<Src: Shape, E: Dtype, D: ReplaceDimKernel<E>, T: Tape<D>> GatherTo<D>
     for Tensor<Src, E, D, T>
 {
-    fn try_gather<Dst: Shape, Idx: Shape>(
+    fn try_gather<Dst: Shape, Idx: Shape, TT: Tape<D>>(
```

Comment:
typo? 😀

Reply:
No theres another T being used by `impl`.

Reply:
Oooooh good call, missed that